lib/src/http/browser/request_mixin.dart

```diff
@@ -45,34 +45,29 @@ abstract class BrowserRequestMixin implements BaseRequest, CommonRequest {
     final c = Completer<BaseResponse>();
 
     // Add request headers.
-    if (finalizedRequest.headers != null) {
-      // The browser forbids setting these two headers:
-      // - connection
-      // - content-length
-      final headersToAdd = Map<String, String>.from(finalizedRequest.headers);
-      headersToAdd.remove('connection');
-      headersToAdd.remove('content-length');
-      headersToAdd.forEach(_request.setRequestHeader);
-    }
+    final headersToAdd = Map<String, String>.from(finalizedRequest.headers);
+    headersToAdd.remove('connection');
+    headersToAdd.remove('content-length');
+    headersToAdd.forEach(_request!.setRequestHeader);
```

Comment:
should we be doing a null check here? if not maybe we want to make this late non-nullable